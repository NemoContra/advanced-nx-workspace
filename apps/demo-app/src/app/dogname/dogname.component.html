<p *ngIf="dogName; else noDogname">{{ greeting }} I am {{ dogName }}</p>
<ng-template #noDogname>
  <p>Please call me by my name!</p>
</ng-template>
