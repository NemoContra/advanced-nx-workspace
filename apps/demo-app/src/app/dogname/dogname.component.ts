import { Component, Input } from '@angular/core';

@Component({
  selector: 'flight-workspace-dogname',
  templateUrl: './dogname.component.html',
  styleUrls: ['./dogname.component.css'],
})
export class DognameComponent {
  @Input() dogName?: string;
  @Input() greeting = 'Wuff';
}
