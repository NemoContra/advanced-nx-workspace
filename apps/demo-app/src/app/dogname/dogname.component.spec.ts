import { DognameComponent } from './dogname.component';
import { createHostFactory } from '@ngneat/spectator/jest';

describe('DognameComponent', () => {
  const createComponent = createHostFactory(DognameComponent);

  it.each`
    dogName      | greeting
    ${'Fido'}    | ${'Wuff wuff'}
    ${'Bello'}   | ${undefined}
    ${'Luna'}    | ${'Raaaaff'}
    ${null}      | ${null}
    ${undefined} | ${'Hey'}
  `(
    'dogname should be $dogname with greeting $greeting',
    ({ dogName, greeting }) => {
      const spectator = createComponent(
        '<flight-workspace-dogname [dogName]="dogName" [greeting]="greeting"></flight-workspace-dogname>',
        {
          hostProps: {
            dogName,
            greeting,
          },
        }
      );
      expect(spectator.fixture).toMatchSnapshot();
    }
  );
});
