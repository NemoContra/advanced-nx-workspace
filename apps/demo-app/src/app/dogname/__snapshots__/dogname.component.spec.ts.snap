// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DognameComponent dogname should be $dogname with greeting Hey 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  dogName="undefined"
  greeting={[Function String]}
>
  <flight-workspace-dogname>
    <p>
      Please call me by my name!
    </p>
    
    
  </flight-workspace-dogname>
</ng-component>
`;

exports[`DognameComponent dogname should be $dogname with greeting Raaaaff 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  dogName={[Function String]}
  greeting={[Function String]}
>
  <flight-workspace-dogname>
    <p>
      Raaaaff I am Luna
    </p>
    
    
  </flight-workspace-dogname>
</ng-component>
`;

exports[`DognameComponent dogname should be $dogname with greeting Wuff wuff 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  dogName={[Function String]}
  greeting={[Function String]}
>
  <flight-workspace-dogname>
    <p>
      Wuff wuff I am Fido
    </p>
    
    
  </flight-workspace-dogname>
</ng-component>
`;

exports[`DognameComponent dogname should be $dogname with greeting null 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  dogName="null"
  greeting="null"
>
  <flight-workspace-dogname>
    <p>
      Please call me by my name!
    </p>
    
    
  </flight-workspace-dogname>
</ng-component>
`;

exports[`DognameComponent dogname should be $dogname with greeting undefined 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  dogName={[Function String]}
  greeting="undefined"
>
  <flight-workspace-dogname>
    <p>
       I am Bello
    </p>
    
    
  </flight-workspace-dogname>
</ng-component>
`;
